kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -488,83 +494,102 @@ define([
+            const data = [];
+
+            if (config.isViewOnly) {
+                let viewData = {
+                    selected: true,
+                    id: config.initialValue || undefined,
+                    text: config.initialValue,
+                };
+                if (config.initialDisplayValue) {
+                    if (typeof config.initialDisplayValue === 'string') {
+                        viewData.text = config.initialDisplayValue;
+                    } else if (
+                        config.initialDisplayValue !== null &&
+                        typeof config.initialDisplayValue === 'object'
+                    ) {
+                        viewData = Object.assign(viewData, config.initialDisplayValue);
+                        if (!viewData.text) {
+                            viewData.text = viewData.id;
                         }
                     }
-                    data.push(viewData);
                 }
-                dropdown.select2({
-                    allowClear: true,
-                    disabled: config.isViewOnly,
-                    templateResult: formatObjectDisplay,
-                    templateSelection: selectionTemplate,
-                    data: config.isViewOnly ? data : null,
-                    ajax: {
-                        delay: 250,
-                        processResults: (data) => {
-                            // update the currently selected one if applicable
-                            const currentData = dropdown.select2('data')[0];
-                            if (currentData) {
-                                const updatedData = data.results.filter((item) => {
-                                    return item.id === currentData.id;
-                                });
-                                if (updatedData.length) {
-                                    for (const key of Object.keys(updatedData[0])) {
-                                        currentData[key] = updatedData[0][key];
-                                    }
-                                    dropdown.trigger('change');
+                data.push(viewData);
+            } else if (model.value && !model.displayValue && ddOptions.exact_match_on) {
+                const searchResults = await fetchData(model.value);
+                // verify that we have an exact match
+                let match = null;
+                for (const result of searchResults) {
+                    if (
+                        model.value.toLowerCase() === result[ddOptions.exact_match_on].toLowerCase()
+                    ) {
+                        match = result;
+                        break;
+                    }
+                }
+                if (match) {
+                    data.push(Object.assign(match, { selected: true }));
+                    model.value = match.id;
+                } else {
+                    model.exactMatchError = true;
+                }
+            }
```

Comment:
Here's the changed block.

Basically, if we have an initial value, we DON'T have a display value yet (which would get populated by the dropdown), and there's an `exact_match_on` field, then we need to do the search. Just use the existing mechanism for that.

If we find a matching result (I'm guessing most of the time there will just be one hit if so), then add the `selected` boolean to it so Select2 knows what to do. Otherwise, trigger the error condition.
This makes it pretty specific to creating a new bulk import cell, as only the uploaders make use of that right now.

If a match gets found,